csharp/ql/src/semmle/code/csharp/frameworks/system/windows/Forms.qll

```diff
@@ -23,3 +23,78 @@ class SystemWindowsFormsHtmlElement extends SystemWindowsFormsClass {
   /** Gets the `SetAttribute` method. */
   Method getSetAttributeMethod() { result = this.getAMethod("SetAttribute") }
 }
+
+/** The `System.Windows.Forms.TextBoxBase` class. */
+class SystemWindowsFormsTextBoxBase extends SystemWindowsFormsClass {
+  SystemWindowsFormsTextBoxBase() {
+    this.hasName("TextBoxBase")
+  }
+
+  /** Gets the `Text` property. */
+  Property getTextProperty() { result = this.getProperty("Text") }
+}
+
+/** The `System.Windows.Forms.RichTextBox` class. */
+class SystemWindowsFormsRichTextBox extends SystemWindowsFormsClass {
+  SystemWindowsFormsRichTextBox() {
+    this.hasName("RichTextBox")
+  }
+
+  /** Gets the `Rtf` property. */
+  Property getRtfProperty() { result = this.getProperty("Rtf") }
+
+  /** Gets the `SelectedText` property. */
+  Property getSelectedTextProperty() { result = this.getProperty("SelectedText") }
+
+  /** Gets the 'SelectedRtf' property. */
+  Property getSelectedRtfProperty() { result = this.getProperty("SelectedRtf") }
+}
+
+/** The `System.Windows.Forms.HtmlDocument` class. */
+class SystemWindowsFormsHtmlDocumentClass extends SystemWindowsFormsClass {
+  SystemWindowsFormsHtmlDocumentClass() {
+    this.hasName("HtmlDocument")
+  }
+
+  /** Gets the `Write` method. */
+  Method getWriteMethod() { result = this.getAMethod() and result.hasName("Write") }  
+}
+
+/** The `System.Windows.Forms.WebBrowser` class. */
+class SystemWindowsFormsWebBrowserClass extends SystemWindowsFormsClass {
+  SystemWindowsFormsWebBrowserClass() {
+    this.hasName("WebBrowser")
+  }
+
+  /** Gets the `DocumentText` property. */
+  Property getDocumentTextProperty() { result = this.getProperty("DocumentText") }
+}
+
+private class TextProperty extends Property {
+  TextProperty() {
+    exists(SystemWindowsFormsRichTextBox c |
+      this = c.getRtfProperty() or
+      this = c.getSelectedTextProperty() or
+      this = c.getSelectedRtfProperty()
+    )
+    or
+    exists(SystemWindowsFormsTextBoxBase tb |
+      this = tb.getTextProperty().getAnOverrider*()
+    )
+  }
+}
+
+/** A field that contains a text control. */
```

Comment:
Why do we restrict this to fields, would it be uncommon to e.g. pass a control as a parameter to a method?